Sage/carbon
# feat(modal-manager): enable global management of modal lists

fix #4170

### Proposed behaviour


Updates `ModalManager` to include mechanism for globally  managing modal lists across multiple versions of carbon

![modal manager working](https://user-images.githubusercontent.com/44157880/155711319-e6003784-76fe-47a7-8b5a-5f6b9f607608.gif)

### Current behaviour


`ModalManager` has not support for maintaining a list of modals that aren't from a unified version of carbon

![modal manager not working](https://user-images.githubusercontent.com/44157880/155711519-e7d6d6e9-dfc2-4695-870b-0ef4cb6246e7.gif)

### Checklist




#### QA


### Additional context



### Testing instructions


* Clone https://github.com/jamime/multiple-carbon-versions
* checkout branch `FE-4185`
* Copy `carbon-react` [url](https://codesandbox.io/s/carbon-quickstart-j8xu0x?file=/package.json:140-144) from package.json in codesandbox build
* Replace the `carbon-react` and `carbon-react-copy` with the url you copied
* Run `npm install && npm start`
* Check that the Confirm component opens for both `external` and `internal` implementations
